programs/ziti-edge-tunnel/package/deb/postinst.in

```diff
@@ -30,8 +30,10 @@ if [ "$1" = "configure" ]; then
   chmod 0770 "@ZITI_IDENTITY_DIR@"
   find "@ZITI_IDENTITY_DIR@" -maxdepth 1 -name "*.json" -type f -exec chown ziti:ziti "{}" + -exec chmod 0400 "{}" +
 
-  policykit_version=$(dpkg-query -Wf '${Version;5}' policykit-1 | cut -d . -f 2)
-  systemd_version=$(dpkg-query -Wf '${Version;3}' systemd)
+  # use minor version as major version, ignoring the magic zero major version
+  policykit_version=$(dpkg-query -Wf '${Version}' policykit-1 | sed -E 's/^([0-9]+\.)?([0-9]+).*/\2/')
+  # use major version
+  systemd_version=$(dpkg-query -Wf '${Version}' systemd | sed -E 's/^([0-9]+)\..*/\1/')
```

Comment:
btw `sort -V` deals with dots and hyphens, and lets you capture the version from dpkg without making any assumptions.

```bash
systemd_version=$(dpkg-query -Wf '${Version}' systemd

if [ $(printf "%s\n" "0.106" "${systemd_version}" | sort -V | head -n1) = "0.106" ]; then
    echo "good enough"
else
    echo "not good enough"
fi
```

Reply:
I like the idea of outsourcing the version parsing to `sort -V.`